lib/active_merchant/connection.rb

```diff
@@ -48,6 +48,11 @@ def initialize(endpoint)
       @proxy_port = nil
     end
 
+    def wiredump_device=(device)
+      raise ArgumentError, "can't wiredump to frozen #{device.class}" if device && device.frozen?
```

Comment:
I think we should make sure that device responds to `frozen?`

Reply:
It's defined on Object so that's pretty much a given? https://ruby-doc.org/core-2.4.2/Object.html#method-i-frozen-3F

Reply:
👍